File: sdks/python/examples/basic_selection.py
"""Basic selection example using the ReinforceSpec SDK."""

import asyncio

from reinforce_spec_sdk import ReinforceSpecClient, SelectionMethod


async def main() -> None:
    async with ReinforceSpecClient(
        base_url="https://api.reinforce-spec.dev",
        api_key="your-api-key",
    ) as client:
        response = await client.select(
            candidates=[
                {"content": "First LLM output", "source_model": "gpt-4"},
                {"content": "Second LLM output", "source_model": "claude-3"},
            ],
            selection_method=SelectionMethod.HYBRID,
            description="Compare LLM outputs for quality",
        )

        print(f"Selected candidate: {response.selected.index}")
        print(f"Composite score: {response.selected.composite_score:.2f}")
        print(f"Confidence: {response.selection_confidence:.2f}")

        for score in response.selected.dimension_scores:
            print(f"  {score.dimension}: {score.score:.1f} ({score.confidence:.0%})")


if __name__ == "__main__":
    asyncio.run(main())

File: sdks/typescript/examples/basic-selection.ts
/**
 * Basic selection example using the ReinforceSpec SDK.
 *
 * Usage:
 *   npx ts-node examples/basic-selection.ts
 */

import { ReinforceSpecClient, SelectionMethod } from '@reinforce-spec/sdk';

async function main() {
  const client = new ReinforceSpecClient({
    baseUrl: 'https://api.reinforce-spec.dev',
    apiKey: 'your-api-key',
  });

  try {
    const response = await client.select({
      candidates: [
        { content: 'First LLM output', sourceModel: 'gpt-4' },
        { content: 'Second LLM output', sourceModel: 'claude-3' },
      ],
      selectionMethod: SelectionMethod.Hybrid,
      description: 'Compare LLM outputs for quality',
    });

    console.log(`Selected candidate: ${response.selected.index}`);
    console.log(`Composite score: ${response.selected.compositeScore.toFixed(2)}`);
    console.log(`Confidence: ${response.selectionConfidence.toFixed(2)}`);

    for (const score of response.selected.dimensionScores) {
      console.log(`  ${score.dimension}: ${score.score.toFixed(1)} (${(score.confidence * 100).toFixed(0)}%)`);
    }
  } finally {
    client.close();
  }
}

main().catch(console.error);
